### PR TITLE
Core/PacketIO: Implement CMSG_UI_MAP_QUEST_LINES_REQUEST

### PR DESCRIPTION
**Pull request by osleyder85**
**Changes proposed:**

-  
-  
-  

**Issues addressed:**

Closes #  (insert issue tracker number)

**Tests performed:**

(Does it build, tested in-game, etc.)

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] 
- [ ]

### CODE REVIEW

**Comment by MaxtorCoder** [comment]
Db2 isn't loaded?

**Comment by osleyder85** [comment]
> Db2 isn't loaded?

Db2 uploaded by Shauren in this commit 5134622a75bdc8ffafb0d86e0c0275d731385d0c

File: src/server/game/Server/Packets/QuestPackets.cpp [hunk 1]
```diff
@@ -772,5 +772,20 @@ void ChoiceResponse::Read()
     _worldPacket >> ResponseIdentifier;
     IsReroll = _worldPacket.ReadBit();
 }
+WorldPacket const* UiMapQuestLinesResponse::Write()
+{
+    _worldPacket << int32(UiMapID);
+    _worldPacket << uint32(QuestLineXQuestIDs.size());
+
+    for (uint32 questIDs : QuestLineXQuestIDs)
+        _worldPacket << questIDs;
```
Comment (MaxtorCoder) [inline src/server/game/Server/Packets/QuestPackets.cpp:781]:
```suggestion
    for (uint32 const& questID : QuestLineXQuestIDs)
        _worldPacket << uint32(questID);
```

File: src/server/game/Handlers/QuestHandler.cpp [hunk 2]
```diff
@@ -867,3 +867,21 @@ void WorldSession::HandlePlayerChoiceResponse(WorldPackets::Quest::ChoiceRespons
             _player->GetReputationMgr().ModifyReputation(sFactionStore.AssertEntry(faction.Id), faction.Quantity);
     }
 }
+void WorldSession::HandleUiMapQuestLinesRequest(WorldPackets::Quest::UiMapQuestLinesRequest& uiMapQuestLinesRequest)
+{
+    UiMapEntry const* uiMap = sUiMapStore.LookupEntry(uiMapQuestLinesRequest.UiMapID);
+
+    if (!uiMap)
```
Comment (MaxtorCoder) [inline src/server/game/Handlers/QuestHandler.cpp:874]:
```suggestion
    UiMapEntry const* uiMap = sUiMapStore.LookupEntry(uiMapQuestLinesRequest.UiMapID);
    if (!uiMap)
```

File: src/server/game/Handlers/QuestHandler.cpp [hunk 3]
```diff
@@ -867,3 +867,21 @@ void WorldSession::HandlePlayerChoiceResponse(WorldPackets::Quest::ChoiceRespons
...
+        return;
+
+    WorldPackets::Quest::UiMapQuestLinesResponse response;
+    response.UiMapID = uiMapQuestLinesRequest.UiMapID;
+
+    for (QuestLineXQuestEntry const* QuestLineXQuest : sQuestLineXQuestStore)
+    {
+        if(QuestLineXQuest->OrderIndex == 0)
+            response.QuestLineXQuestIDs.push_back(QuestLineXQuest->ID);
+    }
```
Comment (MaxtorCoder) [inline src/server/game/Handlers/QuestHandler.cpp:884]:
```suggestion
    for (QuestLineXQuestEntry const* questLineXQuest : sQuestLineXQuestStore)
    {
        if (questLineXQuest->OrderIndex == 0)
            response.QuestLineXQuestIDs.push_back(questLineXQuest->ID);
    }
```

**Review by MaxtorCoder — CHANGES_REQUESTED** [review]

**Comment by Shauren** [comment]
This is not good at all, you are always sending all questlines on all maps, ignoring the uimap

**Pull request closed without merging**